internal/image/image.go

```diff
@@ -48,7 +50,104 @@ func Load(appname string, services []string) error {
 		cmd := exec.Command("docker", "load", "-i", filepath.Join(appname, "images", i))
 		output, err := cmd.CombinedOutput()
 		if err != nil {
-			fmt.Println(output)
+			fmt.Println(string(output))
+			return err
+		}
+	}
+	return nil
+}
+
+func serviceByName(config *composetypes.Config, name string) *composetypes.ServiceConfig {
+	for _, s := range config.Services {
+		if s.Name == name {
+			return &s
+		}
+	}
+	return nil
+}
+
+// ChangeAllImages adds given registry to all images used in config
+func ChangeAllImages(config *composetypes.Config, registry string) error {
+	for i, s := range config.Services {
+		ni, err := ChangeImageRepository(s.Image, registry)
+		if err != nil {
+			return err
+		}
+		config.Services[i].Image = ni
+	}
+	return nil
+}
+
+// ChangeImageRepository changes the registry for image to given value
+func ChangeImageRepository(image, registry string) (string, error) {
+	named, err := reference.ParseNormalizedNamed(image)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse image reference: %s", err)
+	}
+	path := reference.Path(named)
+	if tagged, ok := named.(reference.Tagged); ok {
+		path = path + ":" + tagged.Tag()
+	}
+	path = registry + "/" + path
+	return path, nil
+}
+
+// Push loads and pushes images found in app to given registry
+func Push(appPath string, registry string, services []string, config *composetypes.Config) error {
+	imageDir, err := os.Open(filepath.Join(appPath, "images"))
+	if err != nil {
+		return fmt.Errorf("no images found in app")
+	}
+	defer imageDir.Close()
+	images, err := imageDir.Readdirnames(0)
+	if err != nil {
+		return err
+	}
+	for _, i := range images {
+		if len(services) != 0 && !slices.ContainsString(services, i) {
+			continue
+		}
+		cmd := exec.Command("docker", "load", "-q", "-i", filepath.Join(appPath, "images", i))
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			fmt.Println(string(output))
+			return err
+		}
+		// output is one of:
+		// Loaded image ID: sha256:cf4d5c8667efcb194163da2da2f3da583602bbd9aa5894e785f51c7f5e91bffb
+		// Loaded image: REF
+		soutput := string(output)
+		var ref string
+		if strings.Contains(soutput, "Loaded image:") {
```

Comment:
This part is a little bit fragile as it may change without warning on CLI side.

Reply:
Good point… we may want to reuse `docker/cli`'s code here instead of shelling out

Reply:
Checked the cli code, won't help: the outputed string is what the docker daemon replies.